homeassistant/components/huum/climate.py

```diff
@@ -0,0 +1,131 @@
+"""Support for Huum wifi-enabled sauna."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from huum.const import SaunaStatus
+from huum.exceptions import SafetyException
+from huum.huum import Huum
+from huum.schemas import HuumStatusResponse
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import ClimateEntityFeature, HVACMode
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Huum sauna with config flow."""
+    _huum_handler = Huum(
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+        session=async_get_clientsession(hass),
+    )
+
+    try:
+        entry_status = await _huum_handler.status_from_status_or_stop()
+    except Exception as status_error:
+        raise PlatformNotReady(status_error) from status_error
+
+    async_add_entities([HuumDevice(entry_status, _huum_handler)])
+
+
+class HuumDevice(ClimateEntity):
+    """Representation of a heater."""
+
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_max_temp = 110
+    _attr_min_temp = 40
+
+    _target_temperature = 40
```

Comment:
I think target temp should be None if the sauna is turned off?

Reply:
I don't get it. If it's not heating why do you want to have a target temp?
Still think you should set it to None when the Sauna is off.

Reply:
Even if the sauna is off, it will still have a target temperature. Same as a thermostat, if you turn it off and then on again, will heat to the same temperature as it has been set to before, hence, the target temperature.
So if the sauna was set to 80 degrees, and then turned off, the target temperature if still 80, but it will just not try to reach that until it is turned on.

Reply:
If the API works as it should and there is no issues with it, the sauna will get a target temperature from the API as well. So the 40 is just the lowest value possible and if we for some reason do not get a value from the API, we set it to 40 just to have something there, because that value should always be an integer if we get a response from the API. The return value from HUUMs API is literally called `targetTemperature` as well.